stdlib/list.ml

```diff
@@ -430,3 +430,15 @@ let sort_uniq cmp l =
   in
   let len = length l in
   if len < 2 then l else sort len l
+
+(** {6 Iterators} *)
+
+let to_iter l =
+  let next = function
+    | [] -> Iter.Done
+    | x :: tail -> Iter.Yield (x, tail)
+  in Iter.Sequence (l, next)
+
+let of_iter i =
+  Iter.fold_left (fun acc x -> x::acc) [] i
+  |> rev
```

Comment:
Same comments as for arrays.  One could avoid useless allocations (at least for small enough lists) by storing elements on the stack.